lib/src/http/http_body.dart

```diff
@@ -132,10 +132,12 @@ class HttpBody extends BaseHttpBody {
   /// Returns this request/response body as a JSON object - either a `Map` or a
   /// `List`.
   ///
-  /// This attempts to read this request/response body as a `String` and decode
-  /// it to a JSON object. Throws a [FormatException] if this request/response
-  /// body cannot be decoded to text or if the text is not valid JSON.
-  dynamic asJson() => json.decode(asString());
+  /// This attempts to read this request/response body as a String and decode it
+  /// to a JSON object. If no encoding is specified, it will use UTF-8, rather
+  /// than the generic fallback from the response. Throws a [FormatException] if this
+  /// request/response body cannot be decoded to text or if the text is not
+  /// valid JSON.
+  dynamic asJson() => json.decode((_encoding ?? utf8).decode(asBytes()));
```

Comment:
This seems to break browsers, which already have the string as `xhr.responseText`, so `asBytes` is converting to latin1 and then this code is attempting to decode with utf8.  Regardless, it seems like string->bytes->string round-trip is undesirable overhead.  Perhaps something like this?
```
String asString() => _asString(encoding);
String _asString(Encoding encoding) { ... } // same as current (using param)
dynamic asJson() => json.decode(_asString(_encoding ?? utf8));
```

Reply:
good catch! @alanknight-wk mind making that update?

Reply:
Oops. Missed this. Done, but I changed the parameter name so as not to shadow the getter.